src/cmdstan/command.hpp

```diff
@@ -869,16 +1137,16 @@ int command(int argc, const char *argv[]) {
 
     if (algo->value() == "fullrank") {
       return_code = stan::services::experimental::advi::fullrank(
-          model, *init_context, random_seed, id, init_radius, grad_samples,
-          elbo_samples, max_iterations, tol_rel_obj, eta, adapt_engaged,
-          adapt_iterations, eval_elbo, output_samples, interrupt, logger,
-          init_writer, sample_writer, diagnostic_writer);
+          model, *(init_contexts[0]), random_seed, id, init_radius,
+          grad_samples, elbo_samples, max_iterations, tol_rel_obj, eta,
+          adapt_engaged, adapt_iterations, eval_elbo, output_samples, interrupt,
+          logger, init_writers[0], sample_writers[0], diagnostic_writers[0]);
     } else if (algo->value() == "meanfield") {
       return_code = stan::services::experimental::advi::meanfield(
-          model, *init_context, random_seed, id, init_radius, grad_samples,
-          elbo_samples, max_iterations, tol_rel_obj, eta, adapt_engaged,
-          adapt_iterations, eval_elbo, output_samples, interrupt, logger,
-          init_writer, sample_writer, diagnostic_writer);
+          model, *(init_contexts[0]), random_seed, id, init_radius,
+          grad_samples, elbo_samples, max_iterations, tol_rel_obj, eta,
+          adapt_engaged, adapt_iterations, eval_elbo, output_samples, interrupt,
+          logger, init_writers[0], sample_writers[0], diagnostic_writers[0]);
     }
   }
   stan::math::profile_map &profile_data = get_stan_profile_data();
```

Comment:
The interaction of the multi chain stuff is probably not great with the profiling, right? At the moment you get a single profile.csv where multiple thread id's are listed.... or is this exactly what one wants? Maybe @rok-cesnovar has some thoughts? I'd guess we can go with this as it is presumably.

Reply:
I think a single profile CSV file is what I would want to see. Its less files to deal with and all the info is in one place.

As I said in previous discussions, profiling is something one would typically run with a single chain, as there is no added information gained when running multiple chains. It just adds more noise to the per-gradient numbers. 

So I would keep this simple.